cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/MainApp.scala

```diff
@@ -141,6 +146,37 @@ object MainApp extends ZIOAppDefault {
 
       _ <- preMigrations
       _ <- migrations
+      appConfig <- ZIO.service[AppConfig].provide(SystemModule.configLayer)
+      messagingServiceLayer <-
+        if (appConfig.agent.kafka.enabled) {
+          ZIO.succeed(
+            ZKafkaMessagingServiceImpl.layer(List("kafka:9092"))
+          )
+        } else {
+          ZIO.succeed(
+            InMemoryMessagingService.messagingServiceLayer
+          )
+        }
+      messageProducerLayer <-
+        if (appConfig.agent.kafka.enabled) {
+          ZIO.succeed(
+            ZKafkaProducerImpl.layer[UUID, WalletIdAndRecordId]
+          )
+        } else {
+          ZIO.succeed(
+            InMemoryMessagingService.producerLayer[UUID, WalletIdAndRecordId]
+          )
+        }
+      syncDIDStateProducerLayer <-
+        if (appConfig.agent.kafka.enabled) {
+          ZIO.succeed(
+            ZKafkaProducerImpl.layer[WalletId, WalletId]
+          )
```

Comment:
Good for now, but I would move this bootstrapping logic to the layer that provides either Kafka or InMemory layer to keep the MainApp code shorter

Reply:
That's a good catch. It was not elegant at all, indeed. I refactored and simplified how messaging service and consumer/producer layers can be created, and move the above logic to the MessagingService companion object 👍